src/libraries/System.DirectoryServices/src/System/DirectoryServices/DirectorySearcher.cs

```diff
@@ -195,7 +196,7 @@ public TimeSpan ClientTimeout
         /// Gets or sets the Lightweight Directory Access Protocol (LDAP) filter string format.
         /// </devdoc>
         [DefaultValue(defaultFilter)]
-        public string Filter
+        public string? Filter
```

Comment:
I think it shouldn't be nullable as it has defaulted to non-null and the setter is allowing null but also defaulting if the value was null, it will never be null unless set to null by constructor which might be mistake
```suggestion
        [AllowNull]
        public string Filter
```

Reply:
I've initially had it this way but this can currently be null (maybe due to bug, maybe intentional):

- `public DirectorySearcher(DirectoryEntry? searchRoot, string? filter, string[]? propertiesToLoad, SearchScope scope)` is assigning filter directly and does not do any validation
- in Utils.cs: `internal static Hashtable GetValuesWithRangeRetrieval(DirectoryEntry searchRootEntry, string? filter, ArrayList propertiesWithRangeRetrieval, ArrayList propertiesWithoutRangeRetrieval, SearchScope searchScope)` is being called with explicit null from DirectoryServer.cs `GetParitions()` (call goes through `internal static Hashtable GetValuesWithRangeRetrieval(DirectoryEntry searchRootEntry, string? filter, ArrayList propertiesToLoad, SearchScope searchScope)`)

I'll leave this open and turn this into an issue

Reply:
(likely constructor should assign to property rather than field)

Reply:
https://github.com/dotnet/runtime/issues/49128